### PR TITLE
feat: improved cfn parameters

### PR DESCRIPTION
**Pull request by elonniu**
## Description

- Improved cfn parameters for deployment success rate and efficiency.
- Updated docs.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

### CODE REVIEW

**Review by alvindaiyan — CHANGES_REQUESTED** [review]
pls fix the mis-spellings

**Review by alvindaiyan — APPROVED** [review]
lgtm, gooooooooo

File: infrastructure/src/main.ts [hunk 1]
```diff
@@ -56,6 +56,23 @@ export class Middleware extends Stack {
...
+    });
+
+    const useExist = deployedBefore.valueAsString;
+
+    const s3BucketName = new CfnParameter(this, 'bucket', {
+      type: 'String',
+      description: 'New bucket name or Existing Bucket name',
+      minLength: 3,
+      maxLength: 63,
+      allowedPattern: '^[a-z0-9.-]{3,63}$',
```
Comment (yike5460) [inline infrastructure/src/main.ts:73]: ^(?!xn--|sthree-|sthree-configurator)(?!.*\.\.)[a-z0-9][a-z0-9.-]{1,61}[a-z0-9]$
 refer to https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

**Review by yike5460 — COMMENTED** [review]
comments leaved

File: infrastructure/src/main.ts (hunk 1 above)
Comment (elonniu) [reply infrastructure/src/main.ts:73]: Thank you, since I have merged into the branch, I will update in the new PR.

**Pull request merged**